client/modules/sheaf-core/src/extensions/indent-hack.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file Exports the "indent-hack" for CodeMirror.
+ * This enables elegant, tab-preserving line-wrapping in the editor.
+ */
+
+import { EditorState } from "@codemirror/state"
+import {
+  EditorView,
+  ViewPlugin,
+  ViewUpdate,
+  Decoration,
+  DecorationSet
+} from "@codemirror/view"
+import { RangeSetBuilder } from "@codemirror/rangeset"
+import type { Line } from "@codemirror/text"
```

Comment:
I didn't know this was a thing.

Reply:
Importing types? Yeah, when you have non `tsc` build tools, it's important for them to distinguish between a type import and a, uh, import-import, because it's obviously significantly more complex if you have to resolve imports for their AST.